packages/datatrak-web/src/features/Tasks/CreateTaskModal/CreateTaskModal.tsx

```diff
@@ -200,20 +173,21 @@ export const CreateTaskModal = ({ open, onClose }: CreateTaskModalProps) => {
     }
   }, [selectedCountry?.code]);
 
-  useEffect(() => {
-    if (open) {
-      reset(defaultValues);
-    }
-  }, [open]);
-
   const surveyCode = watch('surveyCode');
 
   return (
-    <Modal isOpen={open} onClose={onClose} title="New task" buttons={buttons} isLoading={isSaving}>
+    <Modal
+      isOpen
+      onClose={onClose}
+      title="New task"
+      buttons={buttons}
+      isLoading={isSaving}
+      disablePortal
+    >
       <Wrapper>
         <LoadingContainer isLoading={isLoadingData} heading="Loading data for project" text="">
           <FormProvider {...formContext}>
```

Comment:
```suggestion

```
With react-hook-form, there is no need to use the FormProvider unless the form spans over multiple components

Reply:
I think here we might need to because one or two fields use `useWatch` from `react-hook-form`

Reply:
I think that you can just use watch from the form like `const { watch } = useForm()`
https://react-hook-form.com/docs/useform/watch

Reply:
Changes pushed 🙏